oxide-auth/src/primitives/registrar.rs

```diff
@@ -189,15 +189,15 @@ impl RegistrarError {
 
 impl Client {
     /// Create a public client.
-    pub fn public(client_id: &str, redirect_uri: Url, default_scope: Scope) -> Client {
```

Comment:
I think this unecessarily breaks the public interface. Also, I do have the concern that clients must have *at least one* redirect `Url`, so I'd rather see instead a added method to provide additional `Url`s. This also makes it clear which is the default uri that gets substituted when the parameter is missing.

```rust
impl Client {
    fn with_alternative_uris(self, uris: Vec<Url>) -> Self;
}
```